File: src/main/java/org/radarbase/management/service/dto/SourceTypeDTO.java
package org.radarbase.management.service.dto;


import java.io.Serializable;
import java.util.HashSet;
import java.util.Objects;
import java.util.Set;
import javax.validation.constraints.NotNull;

/**
 * A DTO for the SourceType entity.
 */
public class SourceTypeDTO implements Serializable {

    private static final long serialVersionUID = 1L;

    private Long id;

    @NotNull
    private String producer;

    @NotNull
    private String model;

    @NotNull
    private String catalogVersion;

    @NotNull
    private String sourceTypeScope;

    @NotNull
    private Boolean canRegisterDynamically = false;

    private String name;

    private String description;

    private String assessmentType;

    private String appProvider;

    private Set<SourceDataDTO> sourceData = new HashSet<>();

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getProducer() {
        return producer;
    }

    public void setProducer(String producer) {
        this.producer = producer;
    }

    public String getModel() {
        return model;
    }

    public String getCatalogVersion() {
        return catalogVersion;
    }

    public void setCatalogVersion(String catalogVersion) {
        this.catalogVersion = catalogVersion;
    }

    public void setModel(String model) {
        this.model = model;
    }

    public String getSourceTypeScope() {
        return sourceTypeScope;
    }

    public void setSourceTypeScope(String sourceTypeScope) {
        this.sourceTypeScope = sourceTypeScope;
    }

    public Set<SourceDataDTO> getSourceData() {
        return sourceData;
    }

    public void setSourceData(Set<SourceDataDTO> sourceData) {
        this.sourceData = sourceData;
    }

    public Boolean getCanRegisterDynamically() {
        return canRegisterDynamically;
    }

    public void setCanRegisterDynamically(Boolean canRegisterDynamically) {
        this.canRegisterDynamically = canRegisterDynamically;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getAssessmentType() {
        return assessmentType;
    }

    public void setAssessmentType(String assessmentType) {
        this.assessmentType = assessmentType;
    }

    public String getAppProvider() {
        return appProvider;
    }

    public void setAppProvider(String appProvider) {
        this.appProvider = appProvider;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }

        SourceTypeDTO sourceTypeDto = (SourceTypeDTO) o;

        if (id == null || sourceTypeDto.id == null) {
            return false;
        }

        return Objects.equals(id, sourceTypeDto.id);
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(id);
    }

    @Override
    public String toString() {
        return "SourceTypeDTO{"
                + "id=" + id
                + ", producer='" + producer + "'"
                + ", model='" + model + "'"
                + ", catalogVersion='" + catalogVersion + "'"
                + ", sourceTypeScope='" + sourceTypeScope + "'"
                + ", canRegisterDynamically='" + canRegisterDynamically + "'"
                + ", name='" + name + '\''
                + ", description=" + description
                + ", appProvider=" + appProvider
                + ", assessmentType=" + assessmentType
                + '}';
    }
}

File: src/main/java/org/radarbase/management/service/dto/SourceTypeDTO.kt
package org.radarbase.management.service.dto

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonSetter
import com.fasterxml.jackson.annotation.Nulls
import java.io.Serializable
import java.util.*
import javax.validation.constraints.NotNull

/**
 * A DTO for the SourceType entity.
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
class SourceTypeDTO : Serializable {
    var id: Long? = null
    lateinit var producer: @NotNull String
    lateinit var model: @NotNull String
    lateinit var catalogVersion: @NotNull String
    lateinit var sourceTypeScope: @NotNull String
    var canRegisterDynamically: @NotNull Boolean = false
    var name: String? = null
    var description: String? = null
    var assessmentType: String? = null
    var appProvider: String? = null

    @set:JsonSetter(nulls = Nulls.AS_EMPTY)
    @JsonInclude(JsonInclude.Include.NON_EMPTY)
    var sourceData: Set<SourceDataDTO> = HashSet()
    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        val sourceTypeDto = other as SourceTypeDTO
        return if (id == null || sourceTypeDto.id == null) {
            false
        } else id == sourceTypeDto.id
    }

    override fun hashCode(): Int {
        return Objects.hashCode(id)
    }

    override fun toString(): String {
        return ("SourceTypeDTO{"
                + "id=" + id
                + ", producer='" + producer + "'"
                + ", model='" + model + "'"
                + ", catalogVersion='" + catalogVersion + "'"
                + ", sourceTypeScope='" + sourceTypeScope + "'"
                + ", canRegisterDynamically='" + canRegisterDynamically + "'"
                + ", name='" + name + '\''
                + ", description=" + description
                + ", appProvider=" + appProvider
                + ", assessmentType=" + assessmentType
                + '}')
    }

    companion object {
        private const val serialVersionUID = 1L
    }
}
